File: HackerRank/projectE2.js
// Problem: Find the sum of the even-valued terms in the Fibonacci sequence whose values do not exceed N.
// Solution: Generate the fibonacci sequence and filter the even terms. Add the total.

function fibonacci_sum(n) {
    // Set initial values.
    let totalSum = 0;
    let previousTerm = 0;
    let currentTerm = 1;

    // Iterate up to n and add all even values of the sequence. 
    while (currentTerm <= n) {
        if (currentTerm % 2 == 0) {
            totalSum += currentTerm;
        }

        // Fibonacci calculation.
        let newTerm = currentTerm + previousTerm;
        previousTerm = currentTerm;
        currentTerm = newTerm;
    }

    // Return sum of even numbers in fibonacci.
    return totalSum;
}

File: HackerRank/projectE2.py
# Problem: Find the sum of the even-valued terms in the Fibonacci sequence whose values do not exceed N.
# Solution: Generate the fibonacci sequence and filter the even terms. Add the total.

# Create fibonacci sequence up to n.
def fibonacci_sum(n):
    # Set initial values.
    current_term = 1
    previous_term = 0
    total_sum = 0
    # Iterate through sequence and add even terms.
    while current_term <= n:
        if current_term % 2 == 0:
            total_sum += new_term
        # fibonacci calculation.
        new_term = previous_term + current_term
        previous_term = current_term
        current_term = new_term
    # Return sum of fibonacci.
    return total_sum


